api/search_test.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"testing"
+
+	"github.com/ONSdigital/dp-nlp-search-scrubber/api/mock"
+	"github.com/ONSdigital/dp-nlp-search-scrubber/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAllMatchingIndustries(t *testing.T) {
+	// get a mock ScrubberDB with some industries
+	mockDB := mock.DB()
+
+	tests := []struct {
+		name          string
+		query         []string
+		expectedCodes []string
+	}{
+		{
+			name:          "matching single query",
+			query:         []string{"ind1"},
+			expectedCodes: []string{"IND1"},
+		},
+		{
+			name:          "matching multiple queries",
+			query:         []string{"ind1", "ind2"},
+			expectedCodes: []string{"IND1", "IND2"},
+		},
+		{
+			name:          "no matching queries",
+			query:         []string{"foo", "bar"},
+			expectedCodes: []string{},
+		},
+		// algorithm of PrefixMap is depth first search
+		// so It will get the data in reverse
+		// keep that in mind when updating tests
+		{
+			name:          "matching partial query",
+			query:         []string{"ind"},
+			expectedCodes: []string{"IND3", "IND2", "IND1"},
+		},
```

Comment:
Although this is a valid test, it's not going to get past your code extraction criteria earlier.

Reply:
Yeah I see what you mean, we can discuss this later on Thursday if you're available? I've got a list of things we can talk about.

Reply:
yeah in params.go the Regexes are looking for codes that are 5 or 8 only. 

So this test is never going to happen in reality because the extraction of the codes from the query param isn't ever going to lead to this input into the matcher.